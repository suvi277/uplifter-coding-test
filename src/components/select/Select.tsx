import React from 'react';
export interface SelectProps {
    allOptions: string[];
    label: string;
}

const Select: React.SFC<SelectProps> = ({ allOptions, label }) => {
    return (
        <div className="input-group input-group-sm mt-3">
            <div className="input-group-prepend">
                <label className="input-group-text">{label}</label>
            </div>
            <select className="custom-select">
                {allOptions.map((tag, index) => (
                    <option key={index} value={tag}>
                        {tag}
                    </option>
                ))}
            </select>
        </div>
    );
};

export default Select;
