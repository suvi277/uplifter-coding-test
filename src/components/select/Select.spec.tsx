import React from 'react';
import { shallow } from 'enzyme';
import Select from './Select';

describe('<Select /> with default props', () => {
    const defaultProps = {
        allOptions: ['a', 'b'],
        label: 'Tags'
    };
    const wrapper = shallow(<Select {...defaultProps} />);
    it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should insert the prop values to correct places ', () => {
        expect(wrapper.find('.input-group-text').text()).toEqual(defaultProps.label);
    });
    it('should render two option tag as allOptions prop length is two ', () => {
        expect(wrapper.find('option').length).toBe(2);
    });
    it('should insert the option values to correct places ', () => {
        expect(wrapper.find('option').at(0).text()).toBe('a');
        expect(wrapper.find('option').at(1).text()).toBe('b');
    });
});
