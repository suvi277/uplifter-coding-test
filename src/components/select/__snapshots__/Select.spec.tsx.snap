// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoCollection /> with default props renders as expected 1`] = `ShallowWrapper {}`;

exports[`<Select /> with default props renders as expected 1`] = `ShallowWrapper {}`;
